mlflow/genai/optimize/optimizers/dspy_mipro_optimizer.py

```diff
@@ -94,11 +95,21 @@ def optimize(
                     requires_permission_to_run=False,
                 )
 
-            return self._format_optimized_prompt(
-                adapter=adapter,
-                program=optimized_program,
-                input_fields=input_fields,
-            )
+        template = self._format_optimized_prompt(
+            adapter=adapter,
+            program=optimized_program,
+            input_fields=input_fields,
+        )
+
+        self._log_optimization_result(optimized_program)
```

Comment:
```suggestion
        self._display_optimization_result(optimized_program)
```

nit: slightly confused by this because of `log_xyz` convention in mlflow.